### PR TITLE
Add org.easyrpg.player

### PR DESCRIPTION
**Pull request by carstene1ns**
### Please confirm your submission meets all the criteria

- [x] Please describe your application briefly.
A modern replacement for an old rpg engine (runs RPG Maker 2000 games everywhere)
- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
`filesystem=home` → user can store games anywhere
`device=all` → needed for gamepad access
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
The linter warnings will be resolved in 0.8.1 release ([upstream PR pending](https://github.com/EasyRPG/Player/pull/3199))

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: org.easyrpg.player.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    sources:
+      - type: git
+        url: https://github.com/fmtlib/fmt.git
+        tag: '9.1.0'
+        commit: a33701196adfad74917046096bf5a2aa0ab0bb50
+    cleanup:
+      - /include
+      - /lib/pkgconfig
+      - /lib/cmake
+  # this is copied from shared-modules to not add the dependency on it
```
Comment (hfiguiere) [inline org.easyrpg.player.yml:50]: why not?

File: org.easyrpg.player.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+      - type: git
+        url: https://github.com/EasyRPG/Player.git
+        tag: '0.8'
+        commit: f8e41f43b619413f95847536412b56f85307d378
+    cleanup:
+      - /share/bash-completion
+      - /share/pixmaps
+    post-install:
+      # Fixup for app id (since rename-* options leave this out)
+      - sed -i 's|easyrpg-player.desktop|org.easyrpg.player.desktop|' /app/share/metainfo/easyrpg-player.metainfo.xml
```
Comment (hfiguiere) [inline org.easyrpg.player.yml:97]: 1. don't use sed for that.
2. this is actually taken care of by flatpak-builder when calling `rename-desktop-file`

Comment (carstene1ns) [reply org.easyrpg.player.yml:97]: Okay, will remove. It did not work while testing, hence the comment.

File: org.easyrpg.player.yml (hunk 1 above)
Comment (carstene1ns) [reply org.easyrpg.player.yml:50]: Just to have no additional submodule step, I would say personal preference.
I could amend this if wanted.

File: org.easyrpg.player.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    sources:
+      - type: git
+        url: https://github.com/fmtlib/fmt.git
+        tag: '9.1.0'
+        commit: a33701196adfad74917046096bf5a2aa0ab0bb50
+    cleanup:
+      - /include
+      - /lib/pkgconfig
+      - /lib/cmake
+      - '*.so'
```
Comment (bbhtt) [inline org.easyrpg.player.yml:35]: Put the common ones in a top level cleanup. Won't need repeating for every module.

File: org.easyrpg.player.yml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      - -DPLAYER_VERSION_APPEND='(Flathub)'
+    sources:
+      - type: git
+        url: https://github.com/EasyRPG/Player.git
+        tag: '0.8'
+        commit: f8e41f43b619413f95847536412b56f85307d378
+    cleanup:
+      - /share/bash-completion
+      - /share/pixmaps
+cleanup:
```
Comment (hfiguiere) [inline org.easyrpg.player.yml:72]: global cleanup is also found above modules.

**Comment by hfiguiere** [comment]
bot, build org.easyrpg.player

File: org.easyrpg.player.yml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  - --device=all # needed for gamepad access
+cleanup:
+  - /include
+  - /lib/cmake
+  - /lib/pkgconfig
+  - /lib/*.so
+modules:
+  # external libraries
+  - shared-modules/linux-audio/fluidsynth2.json
+  - shared-modules/SDL2/SDL2-with-libdecor.json
```
Comment (hfiguiere) [inline org.easyrpg.player.yml:24]: I do believe this is in the runtime since 23.08

**Comment by carstene1ns** [comment]
So, anything missing here?

**Comment by bbhtt** [comment]
> So, anything missing here?

https://github.com/flathub/flathub/pull/5088#discussion_r1543206320

**Comment by bbhtt** [comment]
bot, build org.easyrpg.player

**Pull request closed without merging**